src/setup.jl

```diff
@@ -46,30 +47,43 @@ function vml_prefix(t::DataType)
     error("unknown type $t")
 end
 
+#AbstractArray input check
+memorycheck(A::Array) = true
+memorycheck(A::Base.ReshapedArray) = memorycheck(parent(A))
+memorycheck(A::Base.FastContiguousSubArray) = memorycheck(parent(A))
+memorycheck(A::Base.ReinterpretArray) = memorycheck(parent(A))
```

Comment:
It's possible that this could be achieved by dispatching on `StridedArray`, which includes roughly this list.

Reply:
Well, as IVM only support a part of `StridedArray`, the above check could be done during the compile stage, as
```julia
a = randn(1000)
b = reinterpret(Int,reshape(view(a,1:100),10,10)) # b is not a StridedArray, but valid for IVM
@code_warntype memorycheck(b)
Variables
  #self#::Core.Const(memorycheck)
  A::Base.ReinterpretArray{Int64, 2, Float64, Base.ReshapedArray{Float64, 2, SubArray{Float64, 1, Vector{Float64}, Tuple{UnitRange{Int64}}, true}, Tuple{}}, false}

Body::Bool
1 ─ %1 = Main.parent(A)::Base.ReshapedArray{Float64, 2, SubArray{Float64, 1, Vector{Float64}, Tuple{UnitRange{Int64}}, true}, Tuple{}}
│   %2 = Main.memorycheck(%1)::Core.Const(true)
└──      return %2
```
If we use `StridedArray` to dispatch `memorycheck()`, then we have to check the inputs' `strides` and `size`, which are not available for compile.

Reply:
OK, right it's too broad, since `StridedArray` allows strides other than 1:
```
julia> c10 = collect(1:10.0);

julia> IVM.log!(view(c10, 1:2:10))
5-element view(::Vector{Float64}, 1:2:9) with eltype Float64:
 0.0
 1.0986122886681098
 1.6094379124341003
 7.0
 9.0

julia> c10
10-element Vector{Float64}:
  0.0
  0.6931471805599453
  1.0986122886681098
  1.3862943611198906
  1.6094379124341003
  6.0
  7.0
  8.0
  9.0
 10.0

julia> log.(1:5)'
1×5 adjoint(::Vector{Float64}) with eltype Float64:
 0.0  0.693147  1.09861  1.38629  1.60944
```

Reply:
Well, I just looked into MKL's Api, it seems `BaseStridedVector` is fully supportable.

Reply:
Oh interesting. I wonder if it still speeds them up? Despite the different numbers not being adjacent in memory.